tests/integration/test_task_pool.py

```diff
@@ -47,7 +49,7 @@
         'runahead limit': 'P3',
         'graph': {
             'P1': 'foo & bar',
-            'R1/2': 'foo[1] => pub'  # 2/pub doesn't spawn at start
+            'R1/2': 'foo[1] => pub'
```

Comment:
Can we add a task to this test that doesn't get spawned to the runahead limit? (That was the point of `pub` originally, I think)

Reply:
Yes, I'd like to spend a bit of time making these integration tests more comprehensive.  As I quick fix, I just adapted the existing test to what's actually going on in the scheduler now.

Reply:
New issue: #5012